### DISCUSSION TITLE
Best way to tell if an image is in a shared album?

### DISCUSSION THREAD

**Discussion by gurchik** [RhetTbull/osxphotos#1600] [Q&A]
I'm trying to query my library for photos that are not in a specific Shared Album.

When a photo is added to a Shared Album, the PhotoInfo object in the Shared Album will have a different uuid, filename, path, etc. So what is the best way to determine whether it's in the Shared Album? Currently I'm checking if the `original_filename`, `date`, `description`, and `owner` are the same, but is there a better way?

**Comment by ferid333** [comment]
Hello @gurchik 

To determine whether a photo is in a specific Shared Album, the best approach is to use `osxphotos` and compare the `uuid` of the photo in your library with the `uuid` of photos in the Shared Album. While the `PhotoInfo` object in the Shared Album has a different `uuid`, its metadata fields like `original_filename`, `date`, and `description` might match.

A better approach would be to:
1. Query the Shared Album and extract its photos' metadata (e.g., `original_filename`, `date`, etc.).
2. Compare these against the main library's photos using a combination of `original_filename` and a more robust identifier like `date` (or `fingerprint`, if available).

This avoids solely relying on matching multiple fields, which can be error-prone if metadata changes. Let me know if you'd like an example script!

**Reply by RhetTbull** [reply]
@ferid333 this is a Chat GPT output and not helpful. Please don't try to do engagement farming on this repo.

**Comment by RhetTbull** [comment]
@gurchik sorry I missed this question when you first posted it. There's a value in the Photos database that can be used to map a shared photo back to the original asset but unfortunately this is currently not exposed by OSXPhotos. The value is `ZADDITIONALASSETATTRIBUTES.ZORIGINATINGASSETIDENTIFIER` which maps to the `PhotoInfo.cloud_guid` of the origial asset. I created a simple script that does this as the feature is not currently built into osxphotos. To use it, select one or more photos in Photos then run:

`osxphotos run https://raw.githubusercontent.com/RhetTbull/osxphotos/refs/heads/main/examples/albums.py`

This outputs something like this:

```
Checking albums for selected photo(s)
Found 1 selected photo(s)
Scanning Photos database to match shared images to originals...
Checking photo IMG_0590.heic (3FF2856A-A2A3-4ADF-A8FB-818F0F4EC24D)
Albums:
   Best Photos
Shared Albums:
   Family
```


You can select an original in the library or a photo in a shared album and it will show you all albums and shared albums that photo is part of.
